### PR TITLE
Show list of similar posts

### PR DESCRIPTION
**Pull request by pravintp**

### CODE REVIEW

File: mysite/blog/utils.py [hunk 1]
```diff
@@ -12,3 +15,12 @@ def add_comments_to_post(post, comment):
     new_comment = comment.save(commit=False)
     new_comment.post = post
     new_comment.save()
+
+
+def get_similar_posts(post):
```
Comment (waranlogesh) [inline mysite/blog/utils.py]: Should be part of post

File: mysite/blog/views.py [hunk 2]
```diff
@@ -41,14 +41,16 @@ def post_detail(request, year, month, day, post):
 
     else:
         comment_form = CommentForm()
-
+    
+    similar_posts = get_similar_posts(post)
```
Comment (waranlogesh) [inline mysite/blog/views.py]: Not needed

File: mysite/blog/templates/blog/post/detail.html [hunk 3]
```diff
@@ -13,7 +13,14 @@ <h1>{{ post.title }}</h1>
       Share this post
     </a>
   </p>
-
+  <h2>Similar posts</h2>
+  {% for post in similar_posts %}
```
Comment (waranlogesh) [inline mysite/blog/templates/blog/post/detail.html]: get it from the post directly

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: mysite/blog/models.py [hunk 4]
```diff
@@ -57,6 +58,16 @@ def add_comment(self, comment):
...
         new_comment.save()
 
+    def get_similar_posts(self):
+        post_tags_ids = self.tags.values_list("id", flat=True)
+        similar_posts = Post.published.filter(tags__in=post_tags_ids).exclude(
+            id=self.pk
+        )
+        similar_posts = similar_posts.annotate(same_tags=Count("tags")).order_by(
+            "-same_tags", "-publish"
+        )[:4]
```
Comment (pravintp) [inline mysite/blog/models.py]: simplify

**Review by waranlogesh — APPROVED** [review]

**Pull request merged**